ScatterPeas/scatterpeas/models.py

```diff
@@ -0,0 +1,632 @@
+from datetime import datetime, timedelta
+from pyramid.security import Allow, ALL_PERMISSIONS, Authenticated
+import pytz
+
+from sqlalchemy import (
+    Column,
+    Index,
+    Integer,
+    Text,
+    Unicode,
+    UnicodeText,
+    Table,
+    ForeignKey,
+    String,
+    DateTime,
+    Boolean
+    )
+
+from sqlalchemy import create_engine
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm.exc import NoResultFound
+
+from sqlalchemy.orm import (
+    scoped_session,
+    sessionmaker,
+    relationship,
+    backref
+    )
+
+from cryptacular.bcrypt import BCRYPTPasswordManager
+from zope.sqlalchemy import ZopeTransactionExtension
+
+DATABASE_URL = os.environ.get(
+    'DATABASE_URL',
+    'postgresql:///scatterpeas2'
+)
+
+DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension()))
+
+Base = declarative_base()
+
+
+# engine = create_engine(DATABASE_URL, echo=True)
+# Session = sessionmaker(bind=engine)
+# DBSession = Session()
+
+
+class Reminder(Base):
+    """Reminder table includes payload information for reminders.
+
+    Provides functionality for putting jobs onto the jobs table.
+    Columns include:
+    -id
+    -alias_id: ForeignKey to Alias table
+    -title: Short version of reminder; required
+    -text_payload: To include as email body, optional
+    -media_payload: For including pictures etc. Not yet fully implemented.
+    -rstate: Tracks whether Reminder has pending jobs. True for active.
+    """
+
+    __tablename__ = 'reminders'
+    rrule = relationship("RRule", backref="reminders", uselist=False)
+    jobs = relationship("Job", backref="reminder")
+
+    id = Column(Integer, primary_key=True, nullable=False, autoincrement=True)
+    alias_id = Column(Integer, ForeignKey('aliases.id'), nullable=False)
+    # keeps the rrule currently generating jobs
+    # rrule_id = Column(Integer, ForeignKey('rrules.id'))
+    title = Column(Unicode(256), nullable=False)
+    text_payload = Column(Unicode(20000))
+    # this might end up just being a link to a static resourc; we will see
+    media_payload = Column(Unicode(256))
+    # rstate is used to to track whether a reminder is still producing jobs
+    # turned to False after execution of last job
+    rstate = Column(Boolean)
+
+    @property
+    def __acl__(self):
+        rule_list = []
+        for user in self.alias.users:
+            rule_list.append((Allow, user.username, 'edit'))
+        rule_list.append((Allow, 'group:admin', ALL_PERMISSIONS))
+
+    @classmethod
+    def parse_reminder(cls, reminder_id, session=None):
+        """Takes a reminder and returns the reminder information needed
+        to parse a reminder into a text or email. Also checks for pending jobs,
+        instantiates them. Flips rstate to False if current call is the last"""
+        if session is None:
+            session = DBSession
+        next_job = cls.get_next_job(reminder_id)
+        reminder = cls.retrieve_instance(reminder_id)
```

Comment:
It seems like perhaps `get_next_job` should be an instance method rather than a class method?  I am thinking that you end up getting the specific reminder twice with the API like this, where if it were an instance method you could do this instead:

```python
reminder = cls.retrieve_instance(reminder_id)
next_job = reminder.get_next_job()
```